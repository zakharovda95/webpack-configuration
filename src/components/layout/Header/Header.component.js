import { hyperScript } from "@/helpers/lib/hyper-script.method";

export default class HeaderComponent {

    render() {
        return hyperScript(`
        <div class="header bg-black">
            <div class="header_wrapper">
                <p> Какой то ХЕДЕР! </p>
            </div>
        </div>
        `)
    }

    get Element() {
        return this.render();
    }
}