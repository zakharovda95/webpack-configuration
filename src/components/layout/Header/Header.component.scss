.navbar {
  @media (max-width: 800px) {
    display: flex;
    flex-direction: column;
  }

  min-height: 80px;
  border-bottom: 2px solid lightgray;
  padding: 0 12px;

  display: flex;
  justify-content: space-around;

  .form-inline {
    display: flex;
    gap: 20px;

    @media (max-width: 800px) {
      display: flex;
      flex-direction: column;
      align-items: center;
    }
  }
}
