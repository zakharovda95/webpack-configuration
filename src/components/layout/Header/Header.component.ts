import { hyperScript } from '../../../helpers/lib/hyper-script.method';
import './Header.component.scss';
export class HeaderComponent {
  private readonly header: Element;

  constructor() {
    this.header = this.createElement();
  }

  protected createElement(): Element {
    return hyperScript(`
    <nav class="navbar navbar-expand navbar-dark bg-dark">
        <a class="navbar-brand" href="#">Simple TODO App!</a>
            <form class="form-inline bg-dark">
            <input id="search-input" class="form-control mr-sm-2" type="search" placeholder="Поиск таска">
            
            <div id="dropdown" class="dropdown bg-dark position-relative">
                <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    Сортировка
                </button>
                
                <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
                    <div class="dropdown-item" data-value="newer">Сначала новые</div>
                    <div class="dropdown-item" data-value="older">Сначала старые</div>
                </div>
            </div>
        </form>
        
        <button id="add-btn" class="btn btn-warning my-2 my-sm-0" type="button" data-bs-target="#exampleModalCenter" data-bs-toggle="modal">Новый таск</button>
    </nav>
        `);
  }

  public get Element(): Element {
    return this.header;
  }
}
