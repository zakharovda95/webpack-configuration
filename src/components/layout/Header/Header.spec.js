import { HeaderComponent } from './Header.component';

describe('Header', () => {
  let header;

  beforeEach(() => {
    header = new HeaderComponent();
  });

  test('should be defined', () => {
    expect(header).toBeDefined();
  });

  test('should be instance of HeaderComponent', () => {
    expect(header).toBeInstanceOf(HeaderComponent);
  });

  test('should include all elements', () => {
    expect(header).toMatchSnapshot();
  });
});
