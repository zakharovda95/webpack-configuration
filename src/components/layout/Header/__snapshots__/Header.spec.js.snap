// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should include all elements 1`] = `
HeaderComponent {
  "header": <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    
        
    <a
      class="navbar-brand"
      href="#"
    >
      Simple TODO App!
    </a>
    
            
    <form
      class="form-inline bg-dark"
    >
      
            
      <input
        class="form-control mr-sm-2"
        id="search-input"
        placeholder="Поиск таска"
        type="search"
      />
      
            
            
      <div
        class="dropdown bg-dark position-relative"
        id="dropdown"
      >
        
                
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="btn btn-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          id="dropdownMenuButton"
          type="button"
        >
          
                    Сортировка
                
        </button>
        
                
                
        <div
          aria-labelledby="dropdownMenuButton"
          class="dropdown-menu"
        >
          
                    
          <div
            class="dropdown-item"
            data-value="newer"
          >
            Сначала новые
          </div>
          
                    
          <div
            class="dropdown-item"
            data-value="older"
          >
            Сначала старые
          </div>
          
                
        </div>
        
            
      </div>
      
        
    </form>
    
        
        
    <button
      class="btn btn-warning my-2 my-sm-0"
      data-bs-target="#exampleModalCenter"
      data-bs-toggle="modal"
      id="add-btn"
      type="button"
    >
      Новый таск
    </button>
    
    
  </nav>,
}
`;
